### PR TITLE
feat: expand nonce to 128-bit

### PR DESCRIPTION
**Pull request by zhangsoledad**
### breaking change
* expand nonce to 128-bit
* change pow_message from [nonce + pow_hash] to [pow_hash + nonce ]

### CODE REVIEW

**Comment by zhangsoledad** [comment]
https://github.com/nervosnetwork/ckb-system-scripts/pull/52/files

File: util/jsonrpc-types/src/uint128.rs [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        write!(formatter, "a hex-encoded, 0x-prefixed uint128")
+    }
+
+    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+    where
+        E: Error,
+    {
+        if !value.starts_with("0x") {
+            return Err(Error::custom(format!(
+                "Invalid uint64 {}: without `0x` prefix",
```
Comment (u2) [inline util/jsonrpc-types/src/uint128.rs]: invalid u128?

File: util/jsonrpc-types/src/uint128.rs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                value
+            )));
+        }
+
+        let number = u128::from_str_radix(&value[2..], 16)
+            .map(Uint128)
+            .map_err(|e| Error::custom(format!("Invalid uint64 {}: {}", value, e)))?;
```
Comment (u2) [inline util/jsonrpc-types/src/uint128.rs]: invalid u128?

File: util/jsonrpc-types/src/uint128.rs [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        if number.to_string() != value {
+            return Err(Error::custom(format!(
+                "Invalid uint64 {}: with redundant leading zeros, expected: {}",
```
Comment (u2) [inline util/jsonrpc-types/src/uint128.rs]: invalid u128?

File: spec/src/lib.rs [hunk 4]
```diff
@@ -99,7 +99,7 @@ pub struct Genesis {
     pub compact_target: u32,
     pub uncles_hash: H256,
     pub hash: Option<H256>,
-    pub nonce: u64,
+    pub nonce: u64, // u128 is not supported by https://github.com/alexcrichton/toml-rs
```
Comment (driftluo) [inline spec/src/lib.rs]: We can use `U128` on `numext-fixed-uint` instead, it supports `serde` and accurately express u128

File: sync/src/synchronizer/mod.rs [hunk 5]
```diff
@@ -675,11 +675,11 @@ mod tests {
         let num = 200;
```
Comment (driftluo) [inline sync/src/synchronizer/mod.rs]: let num = 200u128;

Comment (zhangsoledad) [reply sync/src/synchronizer/mod.rs]: this doesn't help, 
in those test will also take the `num` as `BlockNumber`,  cast can't be avoided.
```
fn insert_block(
        chain_controller: &ChainController,
        shared: &Shared,
        nonce: u128,
        number: BlockNumber,
) 
```

**Pull request merged**